tanner/tests/test_mysql_db_helper.py

```diff
@@ -0,0 +1,73 @@
+import unittest
+import asyncio
+import os
+from unittest import mock
+from tanner.utils.mysql_db_helper import MySQLDBHelper
+
+
+def mock_config(section, value):
+    if section == 'SQLI' and value == 'host':
+        return '127.0.0.1'
+    if section == 'SQLI' and value == 'user':
+        return 'root'
+    if section == 'SQLI' and value == 'password':
+        return ''
+
+
+class TestMySQLDBHelper(unittest.TestCase):
+    def setUp(self):
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
+        self.filename = '/tmp/db/test_db'
+        os.makedirs(os.path.dirname(self.filename), exist_ok=True)
+        open('/tmp/db/test_db', 'a').close()
+
+        self.db_name = 'test_db'
+        self.expected_result = None
+        self.returned_result = None
+        self.query_map = None
+        self.handler = MySQLDBHelper()
+
+    @mock.patch('tanner.config.TannerConfig.get', side_effect=mock_config)
+    def test_check_db_exists(self, m):
+        self.expected_result = 0
+
+        async def test():
+            self.returned_result = await self.handler.check_db_exists(self.db_name)
+
+        self.loop.run_until_complete(test())
+
+        if self.returned_result == 1:
+            async def delete():
+                await self.handler.delete_db(self.db_name)
+            self.loop.run_until_complete(delete())
+        else:
+            self.assertEqual(self.expected_result, self.returned_result)
+
+    @mock.patch('tanner.config.TannerConfig.get', side_effect=mock_config)
+    def test_create_query_map(self, m):
+
+        config = {
```

Comment:
Make it more robust. Make at least 2 DBs with each at least 2 tables and different column types. Also make a table name common in the 2 DBs but different column(no of columns/column types)

Reply:
I don't think we should make complicated tests as this might confuse the actual testing function.

Reply:
I don't think it's complicated, at least make it have more real world

Reply:
Okay! I'll give it a shot!

Reply:
https://github.com/mushorg/tanner/pull/315/files#r289592042
What about this test?

Reply:
As, we have added separate test for `setup_db_from_config` so I think no need to do this all again.

Reply:
I don't understand. This methods also executes queries, so how do you whether all DBs, tables are created correctly

Reply:
If the obtained query map is correct as expected then I think all tables are created correctly.

Reply:
So in `test_query_map` the tables obtained are due to  `setup` not because of correct working of `setup_db_from_config`

Reply:
I guess you can use query_map method here to test all the data has been created correctly.

Reply:
> 
> 
> So in `test_query_map` the tables obtained are due to `setup` not because of correct working of `setup_db_from_config`

And it should be like that right? as we are only testing creation of query map.

This should be

Reply:
Don't understand what you said here. But to be clear create_query_map tests are fine and the tests for this method are similar except here you the work of setup is done by setup_from_config method itself.

Reply:
I think you might be referencing old implementation, this implementation you are talking about I have removed in latest commits.

Reply:
Just tell me this how this test is ensuring the tables are formed correctly.

Reply:
Its not. Why it is supposed to? how?

Reply:
Because if a method makes tables then I guess its test should ensure the tables are formed correctly.

Reply:
What about this https://github.com/mushorg/tanner/pull/315/files#discussion_r287583399

Reply:
Yes, tables created in `setup_db_from_config` method are checked.

Reply:
I meant make multiple tables just like you did in createquerymap.

Reply:
Okay

Reply:
> 
> 
> I meant make multiple tables just like you did in createquerymap.

Done.